versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java

```diff
@@ -676,6 +811,14 @@ protected abstract void writeGlobalCommit(
       NonTransactionalOperationContext ctx, GlobalStateLogEntry entry)
       throws ReferenceConflictException;
 
+  /**
+   * Write a new refLog-entry with a best-effort approach to prevent hash-collisions but without any
+   * other consistency checks/guarantees. Some implementations however can enforce strict
+   * consistency checks/guarantees.
+   */
+  protected abstract void writeRefLog(NonTransactionalOperationContext ctx, RefLogEntry entry)
+      throws ReferenceConflictException;
```

Comment:
Thought of having `RefLogConflictException` instead of this. But then all the backend store doesn't throw exception. As reflog writing is common for all the operations, all signatures need to be changed from backend store to REST API If I add new exception.  Hence didn't modified. Any thoughts ?